### DISCUSSION TITLE
Time domain channel

### DISCUSSION THREAD

**Discussion by yhalihal** [NVlabs/sionna#249] [Q&A] [answered]
Hi,

I am in the process of simulating a communication network for single-carrier signals. Initially, I pass my signal through an RRC (Root Raised Cosine) pulse shaping filter, which also serves as an interpolation step. Following this, I intend to pass the signal through a time-domain channel. Upon reviewing the Sionna library, I came across the 'sionna.channel.cir_to_time_channel' function, and I am considering employing either 'sionna.channel.ApplyTimeChannel' or 'sionna.channel.TimeChannel' for this purpose:
https://nvlabs.github.io/sionna/api/channel.wireless.html

However, I encountered a hurdle as these functions presume the utilization of a sinc filter for pulse shaping and receive filtering, whereas my simulation already incorporates an RRC filter for these tasks. Is there an alternative approach or function within the Sionna library that could better align with my scenario, where a different shaping filter is used?

Thank you!

**Comment by faycalaa** [comment, marked as answer]
Hi,

In general, for arbitrary pulse-shaping filters, you would need to compute the convolution of the filtered signal with the channel impulse response. Currently, Sionna does not provide a utility for that.

In the case where a RRC is used on both the transmitter and receiver side, you could implement a function similar to `cir_to_time_channel`, using a raised cosine instead of sinc, to compute the taps. You could then use `ApplyTimeChannel` to apply the computed taps to the transmitted symbols.

**Reply by yhalihal** [reply]
Thank you!
